### PR TITLE
Adding tests for Modular content in search API

### PR DESCRIPTION
**Pull request by ragabala**
This commit takes care of testing the search api returns modular
content, and also checks whether the `updateinfo.xml` has erratum
present in them.

closes #4112

### CODE REVIEW

File: pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py [hunk 1]
```diff
@@ -149,6 +149,43 @@ def test_collection_field(self):
...
+        sync_repo(self.cfg, repo)
+        self.addCleanup(self.client.delete, repo['_href'])
+        repo = self.client.get(repo['_href'], params={'details': True})
+        # Step 2
+        update_info_file = get_repodata(
+            self.cfg,
+            repo['distributors'][0],
+            'updateinfo'
+        )
+        self.assertTrue(update_info_file.findall('.//module'))
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py]: What about we make those assertions a bit more strict? For instance:
Assert that number of module units are per the updateinfo.xml/
Assert that name field, stream, version, context are not empty.

File: pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py [hunk 2]
```diff
@@ -149,6 +149,43 @@ def test_collection_field(self):
...
+        modular_units = search_units(
+            self.cfg, repo, {
+                'filters': {'unit': {'is_modular': True}},
+                'type_ids': ['rpm'],
```
Comment (kersommoura) [inline pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py]: Do we need to search for ``erratum`` unit as well?

**Comment by ragabala** [comment]
Thanks for the reviews @kersommoura . Will work on those.

**Comment by kersommoura** [comment]
Merge of this PR is blocked by https://pulp.plan.io/issues/4477

**Comment by rochacbruno** [comment]
Replacing by #181 

*I tried to push to your branch @ragabala but I guess the `allow contributors to push changes` checkbox is not marked

**Pull request closed without merging**